lib/set.js

```diff
@@ -59,13 +67,12 @@ Set.prototype.addMigration = function(title, up, down){
  */
 
 Set.prototype.save = function(fn){
-  var self = this
-    , json = JSON.stringify(this);
-  fs.writeFile(this.path, json, function(err){
-    if (err) return fn(err);
-
+  var self = this;
+  this.state.save({
```

Comment:
The problem is not with this pull requests, it's with the current functionality of the code. I'm sure that this doesn't break that, but it makes it harder for us to fix since we have to account for all other ways of saving the state.

The problem is if two developers are working in different branches, on different features, and both add one migration each.